models/user_type.go

```diff
@@ -0,0 +1,35 @@
+package models
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// UserType represents the user with email
+type UserType struct {
+	SimpleType
+	value interface{}
+}
+
+func (fieldType UserType) ConvertToModel(value interface{}) (interface{}, error) {
+	if value == nil {
+		return nil, nil
+	}
+	valueType := reflect.TypeOf(value)
+	if valueType.Kind() != reflect.String {
+		return nil, fmt.Errorf("value %v should be %s, but is %s", value, "string", valueType.Name())
+	}
+	// check with DB
```

Comment:
Yes waiting for that part, also I will add related comments around the code. Thanks.